docs/usage.rst

```diff
@@ -82,6 +82,143 @@ Use the batch writer to take care of dynamodb writing retries etc...
 The ``batch_writer()`` can take a keyword argument of ``flush_amount`` which will change the desired flush amount and a keyword argument
 of ``on_exit_loop_sleep``. The ``on_exit_loop_sleep`` argument will add an async sleep in the flush loop when you exit the context manager.
 
+
+S3 Examples
+-----------
+
+Here are some examples of uploading, checking if an S3 object 
+exists and streaming a file from S3, serving via aiohttp.
+
+Upload
+------
+
+Here we upload from a file object and stream it from a file descriptor.
+
+.. code-block:: python3
+
+    async def upload(
+        suite: str,
+        release: str,
+        filename: str,
+        staging_path: Path,
+        bucket: str,
+        aws_secret_access_key: str,
+        aws_access_key_id: str,
+    ) -> str:
+        blob_s3_key = f"{suite}/{release}/{filename}"
+        if not staging_path.exists():
+            LOG.error(
```

Comment:
these docs are going to be displayed in the browser, we can get away with longer lines here so feel free to lengthen them if it aids readability